symengine/parser.cpp

```diff
@@ -196,6 +208,31 @@ class ExpressionParser
                                  parse_string(iter + 1, operator_end[iter]));
                     iter = operator_end[iter] - 1;
 
+                } else if (s[iter] == '=') {
+                    result = Eq(result,
+                                parse_string(iter + 1, operator_end[iter]));
+                    iter = operator_end[iter] - 1;
+
+                } else if (s[iter] == '<' and s[iter + 1] == '=') {
```

Comment:
@srajangarg Though I'm not sure, probably the error occurs in `parse_expr()` where `x >= y` get split into `x >` and the rest. I think it is during the simplification of this expression that the error is thrown up.